src/custom/components/Header/mainMenu/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { useMemo } from 'react'
+import { HeaderLinks as Wrapper, StyledNavLink } from '../styled'
+import { ExternalLink } from 'theme/components'
+import MenuDropdown from 'components/MenuDropdown'
+import { MenuTitle, MenuSection } from 'components/MenuDropdown/styled'
+import SVG from 'react-inlinesvg'
+import { MAIN_MENU, MAIN_MENU_TYPE } from 'constants/mainMenu'
+
+// Assets
+import IMAGE_MOON from 'assets/cow-swap/moon.svg'
+import IMAGE_SUN from 'assets/cow-swap/sun.svg'
+
+const getLinkComponent = {
+  internal: StyledNavLink,
+  external: ExternalLink,
+}
+
+interface LinkType {
+  title: MAIN_MENU_TYPE['title']
+  externalURL?: boolean
+  index: number
+  url: string
+  icon?: string
+  iconSVG?: string
+  handleMobileMenuOnClick: () => void
+}
+
+// Dynamic util function to render links based on internal or external type (with or without icon image)
+function getLink({ title, externalURL, index, url, iconSVG, icon, handleMobileMenuOnClick }: LinkType) {
+  const LinkComponent = externalURL ? getLinkComponent.external : getLinkComponent.internal
```

Comment:
Dynamically define the component, based on if the type of link is `externalURL`

Reply:
Why not `const LinkComponent = externalURL ? StyledNavLink : StyledNavLink` ?

Reply:
Yeah I guess directly referencing works and no need for the extra const.